logic/src/commonMain/kotlin/com/wire/kalium/logic/data/sync/SyncRepository.kt

```diff
@@ -0,0 +1,19 @@
+package com.wire.kalium.logic.data.sync
+
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.updateAndGet
+
+interface SyncRepository {
+    val syncState: StateFlow<SyncState>
+    fun updateSyncState(updateBlock: (currentState: SyncState) -> SyncState): SyncState
+}
+
+class InMemorySyncRepository : SyncRepository {
```

Comment:
What do you think of, maybe marking this repo as `internal` ? So we'll avoid having more instances outside of logic 🤔  I'm not thinking it's blocking, but I was wondering if it could lead to some weird race conditions outside kalium

Reply:
Oooh yeah, absolutely. I'll do that :)